x/stakeibc/keeper/icacallbacks_delegate_test.go

```diff
@@ -0,0 +1,270 @@
+		Amount:			uint64(val1RelAmt),
+	}
+	val2SplitDelegation := types.SplitDelegation{
+		Validator: 		val2.Address,
+		Amount:			uint64(val2RelAmt),
+	}
+	callbackArgs := types.DelegateCallback{
+		HostZoneId:				chainId,
+		DepositRecordId:		depositRecord.Id,
+		SplitDelegations:		[]*types.SplitDelegation{&val1SplitDelegation, &val2SplitDelegation},
+	}
+	args, err := s.App.StakeibcKeeper.MarshalDelegateCallbackArgs(s.Ctx(), callbackArgs)
+	s.Require().NoError(err)
+
+	return DelegateCallbackTestCase{
+		initialState: DelegateCallbackState{
+			stakedBal:		stakedBal,
+			balanceToStake: balanceToStake,
+			depositRecord:	depositRecord,
+			callbackArgs:	callbackArgs,
+			val1Bal: 		val1Bal,
+			val2Bal: 		val2Bal,
+			val1RelAmt:		val1RelAmt,
+			val2RelAmt:		val2RelAmt,
+		},
+		validArgs: DelegateCallbackArgs{
+			packet:   packet,
+			ack: 	  &ack,
+			args:     args,
+		},
+	}
+}
+
+func (s *KeeperTestSuite) TestDelegateCallback_Successful() {
+	tc := s.SetupDelegateCallback()
+	initialState := tc.initialState
+	validArgs := tc.validArgs
+
+	err := stakeibckeeper.DelegateCallback(s.App.StakeibcKeeper, s.Ctx(), validArgs.packet, validArgs.ack, validArgs.args)
+	s.Require().NoError(err)
+
+	// Confirm stakedBal has increased
+	hostZone, found := s.App.StakeibcKeeper.GetHostZone(s.Ctx(), chainId)
+	s.Require().True(found)
+	s.Require().Equal(int64(initialState.stakedBal) + initialState.balanceToStake, int64(hostZone.StakedBal), "stakedBal should have increased")
+
+	// Confirm delegations have been added to validators
+	val1 := hostZone.Validators[0]
+	val2 := hostZone.Validators[1]
+	s.Require().Equal(int64(initialState.val1Bal) + initialState.val1RelAmt, int64(val1.DelegationAmt), "val1 balance should have increased")
+	s.Require().Equal(int64(initialState.val2Bal) + initialState.val2RelAmt, int64(val2.DelegationAmt), "val2 balance should have increased")
+	
+	// Confirm deposit record has been removed
+	records := s.App.RecordsKeeper.GetAllDepositRecord(s.Ctx())
+	s.Require().Len(records, 0, "number of deposit records")
+}
+
+func (s *KeeperTestSuite) TestDelegateCallback_DelegateCallbackTimeout() {
+	tc := s.SetupDelegateCallback()
+	validArgs := tc.validArgs
+	// a nil ack means the request timed out
+	validArgs.ack = nil
+	err := stakeibckeeper.DelegateCallback(s.App.StakeibcKeeper, s.Ctx(), validArgs.packet, validArgs.ack, validArgs.args)
+	s.Require().NoError(err)
+
+	// Confirm stakedBal has not increased
+	hostZone, found := s.App.StakeibcKeeper.GetHostZone(s.Ctx(), chainId)
+	s.Require().True(found)
+	s.Require().Equal(int64(tc.initialState.stakedBal), int64(hostZone.StakedBal), "stakedBal should not have increased")
+
+	// Confirm deposit record has been removed
+	records := s.App.RecordsKeeper.GetAllDepositRecord(s.Ctx())
+	s.Require().Len(records, 1, "number of deposit records")
+	record := records[0]
+	s.Require().Equal(recordtypes.DepositRecord_STAKE, record.Status, "deposit record status should be timeout")
```

Comment:
nit: I think this description might be a little off? 
```suggestion
	s.Require().Equal(recordtypes.DepositRecord_STAKE, record.Status, "deposit record status should not have changed")
```
Few more examples of these in the other tests, if this change is warranted

Reply:
updated them all